server/src/main/scala/viper/gobraserver/GobraServer.scala

```diff
@@ -51,50 +53,61 @@ object GobraServer extends GobraFrontend {
     _server.start()
     VerifierState.flushCachedDiagnostics()
   }
-  
 
-  private def serverExceptionHandling(fileData: FileData, resultFuture: Future[VerifierResult]) {
+  def restart() {
+    _server.stop()
+    ViperBackends.ViperServerBackend.resetServer()
+    init(_options)
+    start()
+  }
+
+  private def serverExceptionHandling(fileData: FileData, resultFuture: Future[VerifierResult]): Future[VerifierResult] = {
```

Comment:
I do not find a (physical) use of this function. Why does it return a Future?

Reply:
I initially experimented with having `restart` return a future such that longer running tasks for stopping and starting could be performed and then `serverExceptionHandling` would return a future that would await a restart (if one happens). At the moment the returned future is only used in `VerificationWorker`